junit-console/src/main/java/org/junit/gen5/console/tasks/XmlReportWriter.java

```diff
@@ -125,6 +126,17 @@ private void writeTestcase(TestIdentifier test, NumberFormat numberFormat, XMLSt
 		writer.writeEndElement();
 	}
 
+	private String getParentClassName(TestIdentifier test) {
+		// @formatter:off
+		return reportData.getTestPlan().getParent(test)
+				.flatMap(TestIdentifier::getSource)
+				.filter(JavaClassSource.class::isInstance)
+				.map(JavaClassSource.class::cast)
+				.map(source -> source.getJavaClass().getName())
+				.orElse("<unrooted>");
```

Comment:
Well, this was the previous code:

```java
writer.writeAttribute("classname", parent.map(TestIdentifier::getName).orElse("<unrooted>"));
```

So I didn't want (and don't want) to change the existing behavior in this PR.

I assumed it was done that way for a reason.

In any case, feel free to raise a new issue if you think this should be changed.